File: w1/mini_max_sum_test.py
"""
Tests for mini_max_sum
"""
from pathlib import Path
from test import testutils
from w1.mini_max_sum import mini_max_sum



class MiniMaxSumTest(testutils.TestCase):
    """
    Mini max sum test cases
    """
    @classmethod
    def setUpClass(cls):
        """Get all the matching test case files"""
        cls.tcs = sorted(Path("w1/tc").glob("2_*.txt"))

    def test_mini_max_sum(self):
        """(W1/2): Mini Max Sum"""
        # iterate over our testcases
        for tc in self.tcs:
            # use a subtest scope
            with self.subTest(tc=tc):
                # read in the lines
                lines = list(self.read_lines(Path(tc)))

                # args: line 1 - but it needs splitting up and turning into a list of ints
                args = [int(value) for value in lines[0].split()]

                # expected: line 3 - also needs splitting up into ints
                expected = [int(value) for value in lines[2].split()]

                # it returns what we expect
                self.assertEqual(mini_max_sum(args), expected)

File: w1/mini_max_sum_test.go
package w1

import (
	"slices"
	"strconv"
	"strings"
	"testing"

	"github.com/treejamie/hackerrank-90days/testutils"
)



func TestMiniMaxSum(t *testing.T){
	// the test cases
	tcs := []string{
		"tc/2_0.txt",
		"tc/2_1.txt",
		"tc/2_14.txt",
	}

	// iterate over the tests
	for _, tc := range tcs {
		// read in the files
		lines, _ := testutils.ReadLines(tc)

		// we want the first line for args
		parts := strings.Fields(lines[0])

		// turn that into a slice of 5 int32
		args := make([]int64, 5)
		for i, arg := range parts{
			n, _ := strconv.ParseInt(arg, 10, 64)
			args[i] = int64(n)
		}

		// parse what we expect to get as a return
		parts = strings.Fields(lines[2])

		// turn that into a slice of 2 int32
		expected := make([]int64, 2)
		for i, arg := range parts{
			n, _ := strconv.ParseInt(arg, 10, 64)
			expected[i] = int64(n)
		}
		//fmt.Printf("%T\n", args)

		// now test
		got := MiniMaxSum(args)
		
		if ! slices.Equal(got, expected) {
			t.Errorf("TestMiniMaxSum: got %d, expected %d", got, expected)
		}

	}

}